workflow/controller/hooks.go

```diff
@@ -11,32 +11,39 @@ import (
 	"github.com/argoproj/argo-workflows/v3/workflow/templateresolution"
 )
 
-func (woc *wfOperationCtx) executeWfLifeCycleHook(ctx context.Context, tmplCtx *templateresolution.Context) error {
+func (woc *wfOperationCtx) executeWfLifeCycleHook(ctx context.Context, tmplCtx *templateresolution.Context) (bool, error) {
+	var hookNodes []*wfv1.NodeStatus
 	for hookName, hook := range woc.execWf.Spec.Hooks {
 		//exit hook will be executed in runOnExitNode
 		if hookName == wfv1.ExitLifecycleEvent {
 			continue
 		}
 		execute, err := argoexpr.EvalBool(hook.Expression, env.GetFuncMap(template.EnvMap(woc.globalParams)))
 		if err != nil {
-			return err
+			return false, err
```

Comment:
According to the [EvalBool](https://github.com/argoproj/argo-workflows/blob/master/util/expr/argoexpr/eval.go) function, the `execute` variable contains a value of 'false' instead of an error when the condition is not met.
Therefore, it does not enter the err condition. 
In addition, [if there is an error, workflows will be treated as failure](https://github.com/GeunSam2/argo-workflows/blob/8573606c16844a10e373f6934210a32f38a8c2a5/workflow/controller/operator.go#L402), so there is no situation of concern.

However, I think it may be confusing as you asked, so I will change the complete value to true even under err conditions.